### ISSUE TITLE
onNodeClick and onNodeDoubleClick both events are not working together 

### ISSUE THREAD

**Issue by sadashiv-sumasoft** [xyflow/xyflow#3959]
### Describe the Bug

When we add the onNodeClick and onNodeDoubleClick event listerns to the ReactFlow component, only onNodeClick is getting fired and onNodeDoubleClick is not even getting trigger.

I have use case where I want to show the some information when we single click on the the node and show some different information when we double click on the node.

Could you please fix this issue or recommentd me some other solutions if you have any.

Thanks in advance. 

### Steps to Reproduce the Bug or Issue

1. Add onNodeClick and onNodeDoubleClick to the react flow component.
2. Try to double click on the node event; it will not get fired.

### Expected behavior

1. Add onNodeClick and onNodeDoubleClick to the react flow component.
2. Try to double click on the node event, event should get trigger.

### Platform

- OS: [e.g.  Windows]
- Browser: [e.g. Chrome]
- Version: [11.10.4]

**Comment by bcakmakoglu** [comment]
[Seems to work fine for me in a sandbox.](https://codesandbox.io/p/devbox/react-flow-starter-ts-forked-gy6nk5?file=%2Fsrc%2FApp.tsx%3A33%2C34&workspaceId=b0b5eb08-9090-4c77-9dc6-14c25ecc472e)

https://github.com/xyflow/xyflow/assets/78412429/37b2c7b0-e1bf-4113-972e-4c201a822e79

Any code you can share that might be the source of the issue?

**Comment by sadashiv-sumasoft** [comment]
In my code for the onNodeClick method, I have added one function that sets one react state. After removing that set state call, it is working fine.
It might be that the set state is an async operation that prevents the onNodeDoubleClicked event from triggering.
Any way I have resolved this issue by onClicking on the node label I am now performing the operations that I wanted to perform on the onNodeDoubleClicked.

Thanks.
